mobx/lib/src/api/observable_collections/observable_list.dart

```diff
@@ -48,34 +48,40 @@ class ObservableList<T>
 
   @override
   int get length {
+    _context.enforceReadPolicy(_atom);
```

Comment:
All calls to `enforeReadPolicy` should also be wrapped in an assert like so:

```dart
assert(() {
  _context.enforeReadPolicy(_atom);
  return true;
}());
```

Otherwise, it won't be properly tree-shaked, especially since `enforeReadPolicy` does more than just an assert.

That's one of the reasons they use `debug` in the name, to make it easier to see that all calls should be wrapped in such assert.

Reply:
Sure, I can do that. Only thing in case of `enforceWritePolicy`, it can't be wrapped as there is a consistency check as well, which is not an assert. If we keep this wrapping inside `enforceReadPolicy`, then for release the method becomes empty with no statement. Will it be tree-shaked to remove the empty method?

We will wrap inside with an `assert`.
```dart
void enforceWritePolicy(Atom atom) {
  assert((){ /* ... */ }());
}
```

During release build, it should become empty:

```dart
void enforceWritePolicy(Atom atom) {
}
```

Will this be tree-shaked as well to remove such empty methods ? If that's the its easier to centralize then to sprinkle the assert wrapper everywhere.

Reply:
I'm not sure about that. It's not very easy to test.
But I know for a fact that flutter wraps both the implementation and the call in an `assert`.

Reply:
Ok, I can apply that wrapper for both read/write. Just for consistency I'll keep it inside the `enforce***` methods. I know it will still result in empty methods but we will not be executing the code at runtime. Hope that's a decent compromise without sacrificing too much readability.

Reply:
That's fine.

Reply:
I've applied the assert wrapped but kept it inside the `enforce***` methods.